[Merged by Bors] - Add cubic curve section to v0.10 release blog post (#565)
bevyengine/bevy-website: content/news/2023-03-04-bevy-0.10/index.md

These abstractions were introduced in [#6404](https://github.com/bevyengine/bevy/pull/6404), [#7381](https://github.com/bevyengine/bevy/pull/7381) and [#7568](https://github.com/bevyengine/bevy/pull/7568).

> ## Cubic Curves

Reviewer: I'd like to see a note on their performance here. It's really impressive!

<div class="release-feature-authors">authors: @aevyrie</div>

> In preparation for UI animation and animation editing, cubic curves have been added to `bevy_math` with a public `CubicGenerator` trait to enable user-defined cubic curves. The implementation provides multiple curves out of the box, useful in various applications:

Reviewer: ```suggestion
In preparation for UI animation and hand-tweaked animation curves cubic curves have been added to `bevy_math` with a public `CubicGenerator` trait to enable user-defined cubic curves. The implementation provides multiple curves out of the box, useful in various applications:
```

> In preparation for UI animation and hand-tweaked animation curves cubic curves have been added to `bevy_math` with a public `CubicGenerator` trait to enable user-defined cubic curves. The implementation provides multiple curves out of the box, useful in various applications:

Reviewer: ```suggestion
In preparation for UI animation and hand-tweaked animation curves, cubic curves have been added to `bevy_math`.  The implementation provides multiple curves out of the box, useful in various applications:
```

Then below discuss the  "public `CubicGenerator` trait to enable user-defined cubic curves".

- `Bezier`: user-drawn splines, and cubic-bezier animation easing for UI - helper methods are provided for cubic animation easing.

> - `Hermite`: useful for smooth interpolation betwen two points in time where you know both the position and velocity, such as network prediction.

Reviewer 2: ```suggestion
- `Hermite`: useful for smooth interpolation between two points in time where you know both the position and velocity, such as network prediction.
```

- `Cardinal`: interpolates between any number of control points, automatically computing tangents; Catmull-Rom is a type of Cardinal spline.

> - `B-Spline`: the only C2 continuous curve, particularly useful for smooth camera motion where continuous acceleration is important.

Reviewer: C2 continuous is very jargony. Even discussing continuity / smoothness of derivatives would be more approachable.

Reviewer: Linking to the wikipedia page would also be okay with me, if you prefer to keep the language.

Author: I ended up removing jargon.

These abstractions were introduced in [#6404](https://github.com/bevyengine/bevy/pull/6404), [#7381](https://github.com/bevyengine/bevy/pull/7381) and [#7568](https://github.com/bevyengine/bevy/pull/7568).

## Cubic Curves

<div class="release-feature-authors">authors: @aevyrie</div>

<video controls loop><source  src="cubic_curves.mp4" type="video/mp4"/></video>
<p class="release-feature-authors">This video shows four kinds of cubic curves being smoothly animated with bezier easing. The curve itself is white, green is velocity, red is acceleration, and blue are the control points that determine the shape of the curve.</p>

In preparation for UI animation and hand-tweaked animation curves, cubic curves have been added to `bevy_math`.  The implementation provides multiple curves out of the box, useful in various applications:

> - `Bezier`: user-drawn splines, and cubic-bezier animation easing for UI - helper methods are provided for cubic animation easing as demonstrated in the above video.

Reviewer: List style is inconsistent here, so CI is failing.
